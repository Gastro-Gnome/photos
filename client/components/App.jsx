import React from 'react';
import Modal from 'react-modal';
import PropTypes from 'prop-types';
import PhotoCarousel from './PhotoCarousel';
import Nav from './Nav';
import PhotosFooter from './PhotosFooter';
// import Modal from './Modal';
import ModalMedia from './ModalMedia';
import styles from './styles/App.css';
import modalStyle from './styles/modalStyle';

Modal.setAppElement(document.getElementById('app'));

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      businessId: props.businessId,
      photos: [],
      currentPhoto: 0,
      modalIsOpen: false,
      modalPhoto: 0,
    };
    this.scroll = this.scroll.bind(this);
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.modalScroll = this.modalScroll.bind(this);
  }

  componentDidMount() {
    // const { businessId } = this.state;
    console.log(window.location.pathname);
    // const businessId = window.location.pathname;
    fetch('photos', {
      method: 'GET',
    }).then(res => res.json())
      .then((res) => {
        console.log(res);
        this.setState({
          photos: res,
          businessId: window.location.pathname,
        });
      })
      .then(() => this.autoScroll())
      .catch(err => console.log(err));
  }

  scroll(direction) {
    const { currentPhoto, photos } = this.state;
    if (direction === 'left' && currentPhoto > 0) {
      this.setState({ currentPhoto: currentPhoto - 1 });
    }
    if (direction === 'right' && currentPhoto < photos.length - 4) {
      this.setState({ currentPhoto: currentPhoto + 1 });
    }
  }

  autoScroll() {
    setInterval(() => this.scroll('right'), 8000);
  }

  openModal(photo) {
    const { photos } = this.state;
    const clickedPhoto = photos.indexOf(photo);
    this.setState({
      modalIsOpen: true,
      modalPhoto: clickedPhoto,
    });
  }

  closeModal() {
    this.setState({ modalIsOpen: false });
  }

  modalScroll(direction) {
    console.log(direction);
    const { modalPhoto, photos } = this.state;
    if (direction === 'left' && modalPhoto > 0) {
      this.setState({ modalPhoto: modalPhoto - 1 });
    }
    if (direction === 'right' && modalPhoto < photos.length - 1) {
      this.setState({ modalPhoto: modalPhoto + 1 });
    }
  }

  render() {
    const {
      photos,
      currentPhoto,
      businessId,
      modalIsOpen,
      modalPhoto,
    } = this.state;
    return (
      <div className={styles.showcaseContainer}>
        <div className={styles.showcaseContainerInner}>
          <PhotoCarousel photos={photos} currentPhoto={currentPhoto} openModal={this.openModal} />
          <Nav direction="nav-left" clickHandler={this.scroll} />
          <Nav direction="nav-right" clickHandler={this.scroll} />
        </div>
        <PhotosFooter quantity={photos.length} businessId={businessId} />
        {/* <Modal
          modalIsOpen={modalIsOpen}
          onRequestClose={this.closeModal}
          modalContent={<div>Hello</div>}
        /> */}
        <Modal
          isOpen={modalIsOpen}
          onRequestClose={this.closeModal}
          style={modalStyle}
        >
          <ModalMedia
            photo={photos[modalPhoto]}
            currentPhoto={modalPhoto}
            quantity={photos.length}
            modalScroll={this.modalScroll}
          />
        </Modal>
      </div>
    );
  }
}

export default App;

App.propTypes = {
  businessId: PropTypes.string,
};
App.defaultProps = {
  businessId: 'archive-bar-and-kitchen-san-francisco',
};
