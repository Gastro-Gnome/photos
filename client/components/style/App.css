.showcaseContainer{
  height: 336px;
  width: 660px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  
}
.showcaseContainerInner{
  height: 220px;
  width: 660px;
}