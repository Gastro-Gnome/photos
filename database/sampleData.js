const sampleData = [
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: '5L-GHeWFLKQvVJj3tvtUDA',
    caption: 'The famous Hearst Pizza!!!',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'bHgmeinxvc5G7MMy6l5hTQ',
    caption: 'Pastrami &amp; Peppers',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'LqjHMjOm3BqJUmtQYD29Hg',
    caption: '',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'Cj479CVSWREChVrirY7hLg',
    caption: 'Shrimp Salad Sando',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'Q6EQkRijW_GlT3OehV7DgQ',
    user_id: 'Z88N6qly3Dp5C06XbZ9IMQ',
    user_photo_id: 'aqRpjUsiZU2o0zERpND-1g',
    caption: 'Interior',
    label: 'inside',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'A83p5DeNA3u3x85QVRmHPg',
    caption: 'Chef Salad',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'ilQ2HM6wDpV67FMtpxCziA',
    user_id: 'yfRPxK0Yq2Xmgw0ioCtHZQ',
    user_photo_id: 'EGUgJrpDKrvvJ0C9fcTJ9A',
    caption: 'Second time = burnt pizza',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'ueGLNXAr1MVKrnYk6av7-g',
    user_id: 'RdWLZfyB5JkQUrTh5NGzWw',
    user_photo_id: 'o5A8_qiYANWoMHOXRqR0bw',
    caption: 'Moscone - seasoned sausage, pepperoni,â€¦',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'uK92G3WUSo381dr2y7oXvg',
    user_id: 'MVQCzaDxDr0JFefXrE38qw',
    user_photo_id: 'pm0eX8F99O3x8M9veewc_g',
    caption: 'Margherita',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'srt3c66dHI5Sxpa8eKVkMQ',
    user_id: 'OPtv-lFAgcddoUuHCcTHUQ',
    user_photo_id: 'LGIxALpwOkQSZk6ZAB6UWg',
    caption: 'Pizza ($17)',
    label: 'food',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'Tv4HHbbcpVgVaPupdlSKJg',
    user_id: 'sBYVrXGv643iR5hbtntvEA',
    user_photo_id: 'exAVFQREsDVUFpqsPqfgwA',
    caption: '',
    label: 'inside',
  },
  {
    business_id: 'archive-bar-and-kitchen-san-francisco',
    photo_id: 'aZPpi_jgFCxO--HGyBpbTQ',
    user_id: '_geqzETHmwyh4TGSHJcOrQ',
    user_photo_id: 'NInuR_ZyzR5hlOBchRidww',
    caption: '',
    label: 'food',
  },

];

export default sampleData;
