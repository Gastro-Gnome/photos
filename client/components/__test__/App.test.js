import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../App';
import sampleData from '../../../database/sampleData';

describe('App', () => {
  beforeEach(() => {
    global.fetch = jest.fn(() => Promise.resolve(sampleData));
  });

  it('renders without crashing', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has a function called scoll that increments and decrements the currentPhoto property in state', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ photos: sampleData });
    const instance = wrapper.instance();
    expect(wrapper.state().currentPhoto).toEqual(0);
    instance.scroll('right');
    expect(wrapper.state().currentPhoto).toEqual(1);
    instance.scroll('left');
    expect(wrapper.state().currentPhoto).toEqual(0);
    instance.scroll('left');
    expect(wrapper.state().currentPhoto).toEqual(0);
    wrapper.setState({ currentPhoto: sampleData.length });
    instance.scroll('right');
    expect(wrapper.state().currentPhoto).toEqual(sampleData.length);
  });
});
