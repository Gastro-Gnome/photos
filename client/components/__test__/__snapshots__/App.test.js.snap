// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendsers without crashing 1`] = `
<div>
  Hello
</div>
`;
